hdrass/CCW+Rotator_integration_testing/Rotator_interlock_test.ipynb

```diff
@@ -0,0 +1,1007 @@
+    "    \"\"\"\n",
+    "    # Eqn (14.1) of Meeus' Astronomical Algorithms\n",
+    "    H = (lst - target.ra).radian\n",
+    "    q = np.arctan2(np.sin(H),\n",
+    "                   (np.tan(location.lat.radian) *\n",
+    "                    np.cos(target.dec.radian) -\n",
+    "                    np.sin(target.dec.radian)*np.cos(H)))*u.rad\n",
+    "    return Angle(q)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_message = \"Interlock Rotator_CCW Integration Test - LVV-T1569\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_message = \"Interlock Rotator_CCW Integration Test - limit switch characterisation\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = salobj.Domain()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "rotation DDS read queue is full (100 elements); data may be lost\n",
+      "motors DDS read queue is full (100 elements); data may be lost\n",
+      "electrical DDS read queue is full (100 elements); data may be lost\n",
+      "ccwFollowingError DDS read queue is full (100 elements); data may be lost\n",
+      "timeAndDate DDS read queue is filling: 15 of 100 elements\n"
+     ]
+    }
+   ],
+   "source": [
+    "script = salobj.Controller(\"Script\", index=1)\n",
+    "rot = salobj.Remote(d, \"MTRotator\")\n",
+    "mtptg = salobj.Remote(d, \"MTPtg\")\n",
+    "mtm = salobj.Remote(d, \"MTMount\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await asyncio.gather(rot.start_task,\n",
+    "                     mtptg.start_task,\n",
+    "                     script.start_task,\n",
+    "                     mtm.start_task)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await asyncio.sleep(1.)\n",
+    "await salobj.set_summary_state(mtptg, salobj.State.ENABLED)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await rot.cmd_clearError.start(timeout=10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await salobj.set_summary_state(rot, salobj.State.DISABLED)                             "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await rot.cmd_start.set_start(timeout=10)"
```

Comment:
Please add a comment explaining what you expect here. I expect MTRotator to go to fault if the MTMount is not in DISABLED or ENABLED state. Is that what you are testing?

You should never fully enable MTRotator unless MTMount is in DISABLED or ENABLED state because MTRotator will go to FAULT (it can't see the rotator telemetry).

Similarly you should never fully enable MTMount unless MTRotator is in DISABLED or ENABLED state, because MTMount's CCW following loop will give up.

I recommend this startup sequence:
- MTRotator to DISABLED (start command)
- MTMount to ENABLED (if the rotator and CCW are far apart, this gives MTMount a chance to center the CCW)
- MTRotator to ENABLED

The next few code blocks seem to have the same issue. Please consider combining all these code blocks into a single one that safely enables the system by executing the commands in the right sequence.

Or if you are trying to intentionally bring things up in the wrong order, code blocks that do that (e.g. by sending MTMount to STANDBY and then enabling MTRotator, which will send it to FAULT) with appropriate explanations.

Reply:
Startup sequence at the beginning of notebook updated